api.go

```diff
@@ -19,18 +19,36 @@ type api struct {
 	// value returns a pointer to a newly allocated Go variable able to
 	// represent the JSON object returned by the query.
 	value func() interface{}
+
+	// args is a struct that that the JSON request body is decoded into.
+	args struct {
+		ConferenceID int `json:"conference_id" db:"conference_id"`
+	}
 }
 
+// TODO: return an empty array instead of nothing if no results returned from database?
+
+// TODO: if args are required but not supplied, return a specific error message?
+
 func (a *api) serve(s *server) {
-	// TODO(mdempsky): Add mechanism for queries to specify custom
-	// arguments picked out of the request.
+	err := json.NewDecoder(s.r.Body).Decode(&a.args)
```

Comment:
You'll want to change args into a `func() interface{}` like for value. As is, this means every request to the same API is modifying the same global variable, which is a race condition.